packages/ui/src/ui/store/selectors/acl.ts

```diff
@@ -0,0 +1,350 @@
+}
+
+function prepareApprovers(
+    approvers: Array<PreparedRole> | undefined,
+    type: 'read_approver' | 'responsible' | 'auditor',
+) {
+    return _.map(approvers, (subject) => {
+        const extra = {
+            type,
+            subjects: [subject.value],
+            subjectType: subject.type === 'users' ? ('user' as const) : ('group' as const),
+            groupInfo:
+                subject.type === 'groups'
+                    ? {name: subject.group_name, url: subject.url, group: undefined}
+                    : undefined,
+            action: undefined,
+        };
+        return {
+            ...subject,
+            ...extra,
+        };
+    });
+}
+
+export type PreparedApprover = ReturnType<typeof prepareApprovers>[number];
+
+export const getAllUserPermissions = (state: RootState, idmKind: IdmKindType) =>
+    state.acl[idmKind].userPermissions;
+const getAllObjectPermissions = (state: RootState, idmKind: IdmKindType) =>
+    state.acl[idmKind].objectPermissions;
+
+const getAllObjectPermissionsWithSplittedSubjects = createSelector(
+    [getAllObjectPermissions],
+    splitSubjects,
+);
+
+export const getObjectPermissionsTypesList = (idmKind: IdmKindType) => {
+    return createSelector(
+        [
+            getObjectPermissionsFilter,
+            (state) => getAllObjectPermissionsWithSplittedSubjects(state, idmKind),
+        ],
+        (permissionsFilter, items) => {
+            const uniquePermisions = new Set<YTPermissionTypeUI>();
+            const {permissionTypes} = UIFactory.getAclPermissionsSettings()[idmKind] || {};
+            [...permissionTypes, ...permissionsFilter].forEach((permission) =>
+                uniquePermisions.add(permission),
+            );
+            items.forEach((item) => {
+                const {permissions} = item;
+                permissions?.forEach((permission) => uniquePermisions.add(permission));
+            });
+            return _.sortBy([...uniquePermisions], (permission) => permission);
+        },
+    );
+};
+
+type HasSplitted = {
+    isSplitted?: boolean;
+    subjectIndex?: number;
+};
+
+function splitSubjects<T extends {subjects: Array<unknown>}>(items: Array<T>) {
+    const res: Array<T & HasSplitted> = [];
+    _.forEach(items, (item) => {
+        const {subjects} = item;
+        if (subjects && subjects.length > 1) {
+            _.forEach(subjects, (subject, index) => {
+                res.push({...item, subjects: [subject], isSplitted: true, subjectIndex: index});
+            });
+        } else {
+            res.push(item);
+        }
+    });
+    return res;
+}
+
+function subjectFilterPredicate<
+    T extends {subjectType?: unknown; groupInfo?: unknown; subjects: Array<unknown>},
+>(item: T, filter: string) {
+    const {subjectType, groupInfo} = item;
+    if (subjectType === 'group') {
+        return _.some(Object.entries(groupInfo ?? {}), ([key, value]) => {
+            let str: string | undefined = String(value);
```

Comment:
As far as I remember, String cannot be undefined

Reply:
`undefined` is required by the expression below:
```ts
    str = str.split('/').pop();
```